tracer/src/Datadog.Trace/AppSec/Rcm/ConfigurationStatus.cs

```diff
@@ -118,6 +116,11 @@ internal static List<RuleData> MergeRuleData(IEnumerable<RuleData> res)
             dictionary.Add(WafRulesDataKey, rulesData.Select(r => r.ToKeyValuePair()).ToArray());
         }
 
+        if (IncomingUpdateState.WafKeysToApply.Contains(WafActionsKey))
+        {
+            dictionary.Add(WafActionsKey, Actions.Select(a => a.Value.ToKeyValuePair()).ToArray());
```

Comment:
This is interesting. Our encoder does not process ToList(). When we do ToArray(), we cast later the result to IList<object> which is not possible in the case of ToList(). It would be interesting to update our encoder and review all the ToArray() that we have in this file to make it more efficient. I will add it as a possible task. Thanks!

(Actions.Select(a => a.Value.ToKeyValuePair()).ToList() as IList<object>)
null

(Actions.Select(a => a.Value.ToKeyValuePair()).ToArray() as IList<object>)
{System.Collections.Generic.List<System.Collections.Generic.KeyValuePair<string, object>>[1]}
    [0]: Count = 3

(Actions.Select(a => a.Value.ToKeyValuePair()).ToArray()).GetType().FullName
"System.Collections.Generic.List`1[[System.Collections.Generic.KeyValuePair`2[[System.String, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e],[System.Object, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]], System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]][]"

(Actions.Select(a => a.Value.ToKeyValuePair()).ToList()).GetType().FullName
"System.Collections.Generic.List`1[[System.Collections.Generic.List`1[[System.Collections.Generic.KeyValuePair`2[[System.String, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e],[System.Object, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]], System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]], System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]]"